subsys/bluetooth/audio/ascs.c

```diff
@@ -2033,30 +2033,57 @@ static bool ascs_parse_metadata(struct bt_data *data, void *user_data)
 		return false;
 	}
 
-	/* The CAP acceptor shall not accept metadata with
-	 * unsupported stream context.
-	 */
-	if (IS_ENABLED(CONFIG_BT_CAP_ACCEPTOR)) {
-		if (data_type == BT_AUDIO_METADATA_TYPE_STREAM_CONTEXT) {
-			const uint16_t context = sys_get_le16(data_value);
+	switch (data_type) {
+	/* TODO: Consider rejecting BT_AUDIO_METADATA_TYPE_PREF_CONTEXT type */
+	case BT_AUDIO_METADATA_TYPE_PREF_CONTEXT:
+	case BT_AUDIO_METADATA_TYPE_STREAM_CONTEXT: {
+		uint16_t context;
+
+		if (data_len != sizeof(context)) {
+			*result->rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_METADATA_INVALID,
+						       data_type);
+			result->err = -EBADMSG;
+			return false;
+		}
+
+		context = sys_get_le16(data_value);
+		if (context == BT_AUDIO_CONTEXT_TYPE_PROHIBITED) {
+			*result->rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_METADATA_REJECTED,
```

Comment:
Is `BT_BAP_ASCS_RSP_CODE_METADATA_REJECTED` the right status code? Doesn't `BT_BAP_ASCS_RSP_CODE_METADATA_REJECTED` indicate that it is reject because we do not want it / support it, where `BT_AUDIO_CONTEXT_TYPE_PROHIBITED` is actually an invalid value?

Reply:
As per TS, the error code can be Rejected Metadata or Invalid Metadata

Reply:
I suggest invalid metadata in this case. At least that makes more sense to me